### PR TITLE
Generic checkout : Tier3  Thank-you pages

### PR DESCRIPTION
**Pull request by paul-daniel-dempsey**
## What are you doing in this PR?

re:Tier3 requirements ->

1. Existing Thankyou page app download remove divider below tablet breakpoint
2. ThankyouPage Header Change
3. Create `Benefits` Tier3 component
5. Create `SubscriptionStart` Tier3 component
6. Create `SignedIn` Tier3 component
7. Amend `SignUp` components with Tier3 content.
8. Amend `SignIn` components with Tier3 content.
9. Remove `ShareSupport` component, Add `AppsDownload` component

[**Trello Card**](https://trello.com/c/japx3mbT/941-generic-checkout-thank-you-page-updates-for-top-tier-checkout)

## How to test
https://support.thegulocal.com/uk/contribute#ab-tierThreeFromApi=variant

## Screenshots

|2|3|4|5|6|7|
|-----|-----|-----|-----|-----|-----|
|image|image|image|image|image|image|

### CODE REVIEW

File: support-frontend/assets/components/thankYou/appDownload/appDownloadItems.tsx
Comment (jamesgorrie) [inline support-frontend/assets/components/thankYou/appDownload/appDownloadItems.tsx:1]: I'm getting quite confused with the structure of these thank you pages... Why is the subscription info in `AppDownload`?

Maybe this is just a legacy of this section losing structure over time - might be worth thinking of maybe a post tidy (I know we've said this before, so might be time)?

File: support-frontend/assets/components/thankYou/signedIn/signedInItems.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import BulletPointedList from 'components/thankYou/utilityComponents/BulletPointedList';
+import type { CsrfState } from 'helpers/redux/checkout/csrf/state';
+import { routes } from 'helpers/urls/routes';
+import { isCodeOrProd } from 'helpers/urls/url';
+import { catchPromiseHandler } from 'helpers/utilities/promise';
+
+const expandableContainer = css`
+	margin-top: ${space[4]}px;
+
+	& > * + * {
```
Comment (jamesgorrie) [inline support-frontend/assets/components/thankYou/signedIn/signedInItems.tsx:17]: Could we leave a note explaining what this is doing?

File: support-frontend/assets/components/thankYou/signedIn/signedInItems.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+	useEffect(() => {
+		if (email) {
+			const payload = {
+				email,
+			};
+
+			fetchSignInLink(payload);
+		}
+	}, []);
```
Comment (jamesgorrie) [inline support-frontend/assets/components/thankYou/signedIn/signedInItems.tsx:91]: assuming `email` should be a dependency?

**Review by jamesgorrie — APPROVED** [review]
This looks good and works! I think we are probably close to pushing the current thank you page structure and wonder if we could gain some benefits from moving over to a more product based page.

I know this is a little difficult given the coupling to the old contribution models, but this is getting quite complicated for what is a relatively simple page.

File: support-frontend/assets/components/thankYou/appDownload/appDownloadItems.tsx
Comment (paul-daniel-dempsey) [reply support-frontend/assets/components/thankYou/appDownload/appDownloadItems.tsx:1]: Should not be there, have moved SubscriptionStart and Benefits out to a seperate thankyou component file now.

File: support-frontend/assets/pages/[countryGroupId]/thank-you.tsx [hunk 3]
```diff
@@ -139,40 +149,55 @@ export function ThankYou({ geoId }: Props) {
...
-		),
-		...maybeThankYouModule(
-			contributionType !== 'ONE_OFF' && isSupporterPlus,
-			'appDownload',
-		),
-		...maybeThankYouModule(
-			contributionType === 'ONE_OFF' && emailExists,
-			'supportReminder',
-		),
+		...maybeThankYouModule(isNewAccount, 'signUp'), // Create your Guardian account
```
Comment (paul-daniel-dempsey) [inline support-frontend/assets/pages/[countryGroupId]/thank-you.tsx:191]: Have tried to simplfy/tidy up the logic here

File: support-frontend/assets/components/thankYou/signedIn/signedInItems.tsx (hunk 1 above)
Comment (paul-daniel-dempsey) [reply support-frontend/assets/components/thankYou/signedIn/signedInItems.tsx:18]: Yes, comment added above selectors.

File: support-frontend/assets/components/thankYou/signedIn/signedInItems.tsx (hunk 2 above)
Comment (paul-daniel-dempsey) [reply support-frontend/assets/components/thankYou/signedIn/signedInItems.tsx]: Your right, its not needed at all if your already signedIn, have simplified the component, removing the need for `email` & associated `csrf` in this component.

**Comment by paul-daniel-dempsey** [comment]
> This looks good and works! I think we are probably close to pushing the current thank you page structure and wonder if we could gain some benefits from moving over to a more product based page.
> 
> I know this is a little difficult given the coupling to the old contribution models, but this is getting quite complicated for what is a relatively simple page.

You mean, adjust `thank-you` to loop through thank-you components for a given product - possibly defined on the `order` object within?  Could add a Trello Card for this if you want?

Or do you (also?) mean replace `SupporterPlusThankyou` + `digitalSubscriptionThankyou` pages with this single `thank-you` earlier than retiring the associated checkouts?  I think these cards will come up soon enough anyway but earlier can't  harm if we have time (stretch goals?).

**Pull request merged**